File: A3/src/stmt/PrintStmt.java
package stmt;

import exp.Exp;
import list.*;
import dictionary.*;
import model.*;

public class PrintStmt implements IStmt
{
	Exp exp;
	public PrintStmt(Exp exp)
	{
		this.exp = exp;
	}
	public PrgState execute(PrgState state) throws MyStmtException
	{
		MyIList<Integer> ot = state.getOut();
		MyIDictionary<String, Integer> symtbl = state.getSymTable();
		int val = exp.eval(symtbl);
		ot.add(val);
		return state;
	}
	public String toString()
	{
		return "print (" + exp.toString() + ")";
	}
}

File: A7/Stmt/PrintStmt.cs
using A7.Dictionary;
using A7.Expr;
using A7.List;
using A7.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Stmt
{
    class PrintStmt: IStmt
    {
        Exp exp;
        public PrintStmt(Exp exp)
        {
            this.exp = exp;
        }
        public PrgState Execute(PrgState state)
        {
            MyIList<int> ot = state.GetOut();
            MyIDictionary<String, int> symtbl = state.GetSymTable();
            int val = exp.Eval(symtbl);
            ot.Add(val);
            return state;
        }

        public override string ToString()
        {
                return "print (" + exp.ToString() + ")";
        }
    }
}
